cpp/grammar_functor.cc

```diff
@@ -344,39 +466,22 @@ class StructureNormalizerImpl : public GrammarMutator {
   }
 };
 
-class ByteStringFuserImpl : public GrammarMutator {
+/*!
+ * \brief A class that normalizes a grammar by applying a series of transformations.
+ *
+ * The normalizer applies the following transformations in order:
+ * 1. SingleElementExprEliminator - Eliminates single element expressions
+ * 2. NestedRuleUnwrapper - Unwraps nested rules
+ */
+class GrammarNormalizerImpl {
```

Comment:
We can remove StructureNormalizerImpl and put its logic into GrammarNormalizerImpl